config/initializers/class_decorator.rb

```diff
@@ -0,0 +1,22 @@
+class ApplicationRecord < ActiveRecord::Base
+  class << self
+    def decorate
+      @_decorator || (@_decorator = decorator_for(self))
+    end
+
+    def decorator_for(klass)
+      decorator = nil
+
+      while decorator.nil? && klass != ApplicationRecord
+        decorator = "#{klass.name}Decorator".safe_constantize
+        klass = klass.superclass
+      end
+
+      decorator
+    end
+  end
+
+  def decorate
+    self.class.decorator_for(self.class).new(self)
+  end
+end
```

Comment:
(Follow this after addressing comment [here](https://github.com/ManageIQ/manageiq-ui-classic/pull/237/files#r97825145))

Patching something that doesn't belong to you is a bad idea, especially re-opening a class in our lovely project 😄 

Instead, you can do away with this entirely and create these methods in `MiqDecorator` itself. `MiqDecorator` can take in instances and klasses and return the proper decorator (which it owns) instance or class for you.

```ruby
class MiqDecorator < SimpleDelegator
  self << class
    def decorator_for(klass)
       "#{klass.name}Decorator".safe_constantize
    end

    def decorate(instance)
      decorator = instance_for(instance.class)
      decorator.new(instance)
    end
  end
end
```

Disclaimer: I just wrote this pseudo code up quick, a few details are plainly missing.